service/application/src/main/java/com/solace/maas/ep/event/management/agent/processor/RouteCompleteProcessorImpl.java

```diff
@@ -32,13 +40,69 @@ public void process(Exchange exchange) throws Exception {
             // Sending Error message to client
             exchange.getIn().setHeader("SCAN_ERROR", constant(true));
         } else {
-            log.info("Route {} completed for scan request: {}.", scanType, scanId);
-            if (scanLifecycleService.scanRouteCompleted(scanId)) {
-                // Just in case we're chaining more stuff at the end, set the header to
-                // show we're done
-                exchange.getIn().setHeader("SCAN_COMPLETE", true);
-                log.info("Scan request {} completed.", scanId);
-            }
+            String scanId = (String) exchange.getIn().getHeader(RouteConstants.SCAN_ID);
+            String scanType = (String) exchange.getIn().getHeader(RouteConstants.SCAN_TYPE);
+            String groupId = (String) exchange.getIn().getHeader(RouteConstants.SCHEDULE_ID);
+            String messagingServiceId = (String) exchange.getIn().getHeader(RouteConstants.MESSAGING_SERVICE_ID);
+
+            ScanStatusEntity scanStatusEntity = ScanStatusEntity.builder()
+                    .scanId(scanId)
+                    .scanType(scanType)
+                    .status(ScanStatus.COMPLETE.name())
+                    .build();
+
+            save(scanStatusEntity);
+
+            sendScanDataStatus(scanId, scanType, groupId, messagingServiceId, ScanStatus.COMPLETE, ScanStatusType.PER_ROUTE);
+            log.info("Route {} completed for scan request {}", scanType, scanId);
         }
     }
+
+    private RouteEntity createScanStatusRoute(String scanType, String messagingServiceId) {
+        MessagingServiceRouteDelegate delegate = PluginLoader.findPlugin("SCAN_STATUS");
```

Comment:
Why is this a plugin? Sending Scan Statuses hasn't been created as a separate plugin module. You either need to create a separate plugin module and make it a plugin, or just leave the code inside of the EMA (Application Module) and not have it as a plugin.